File: abc/abc166/abc166_c.fs
// https://atcoder.jp/contests/abc166/submissions/12783941

let [| N; M |] = stdin.ReadLine().Split() |> Array.map int
let H = stdin.ReadLine().Split() |> Array.map int

let AB = Array.zeroCreate<int> N

for _ in [ 1..M ] do
    let ab = stdin.ReadLine().Split() |> Array.map int
    AB.[ab.[0] - 1] <- max AB.[ab.[0] - 1] H.[ab.[1] - 1]
    AB.[ab.[1] - 1] <- max AB.[ab.[1] - 1] H.[ab.[0] - 1]

AB
|> Array.indexed
|> Array.filter (fun x -> H.[fst x] > snd x)
|> Array.length
|> stdout.WriteLine

File: abc/abc166/abc166_c.cs
// https://atcoder.jp/contests/abc166/submissions/12784359

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var input = Console.ReadLine().Split().Select(int.Parse).ToArray();
        var N = input[0];
        var M = input[1];

        var H = Console.ReadLine().Split().Select(int.Parse).ToArray();

        var AB = new int[N];
        for (int i = 0; i < M; i++)
        {
            input = Console.ReadLine().Split().Select(int.Parse).ToArray();
            var A = input[0];
            var B = input[1];

            AB[A - 1] = Math.Max(AB[A - 1], H[B - 1]);
            AB[B - 1] = Math.Max(AB[B - 1], H[A - 1]);
        }

        var result = 0;
        for (int i = 0; i < N; i++)
        {
            if (H[i] > AB[i])
            {
                result++;
            }
        }
        Console.WriteLine(result);
    }
}
